discovery_engine/lib/src/domain/models/document.dart

```diff
@@ -47,19 +47,32 @@ class Document {
   @HiveField(6)
   DateTime timestamp;
 
+  /// Indicates if this [Document] was returned in response to active search.
+  @HiveField(7)
+  bool isSearched;
+
   bool get isRelevant => userReaction == UserReaction.positive;
   bool get isNotRelevant => userReaction == UserReaction.negative;
   bool get isNeutral => userReaction == UserReaction.neutral;
 
+  /// Returns [NewsResource] snippet, or title if snippet is an empty String.
+  String get snippet =>
+      resource.snippet.isNotEmpty ? resource.snippet : resource.title;
+
   Document({
     required this.stackId,
     required this.resource,
     required this.batchIndex,
-    DocumentId? documentId,
+    required this.documentId,
     this.userReaction = UserReaction.neutral,
     this.isActive = true,
-  })  : documentId = documentId ?? DocumentId(),
-        timestamp = DateTime.now().toUtc();
+    this.isSearched = false,
+  }) : timestamp = DateTime.now().toUtc();
+
+  /// Register a [UserReaction]
+  void registerReaction(UserReaction reaction) {
```

Comment:
the `userReaction` field is public anyways and therefore has a default setter, it might cause more confusion than benefits to have another setter with a different name but same functionality.

i get that you want to call it in the ellipsis
```dart
doc..registerReaction(userReaction)
```
but this can already be achieved by
```dart
doc..userReaction = userReaction
```

or we could call it like the following (imo a bit easier to read)
```dart
doc.userReaction = userReaction;
doc
```

Reply:
also 'register' is a bit weird I would expect to save it somewhere not only assign it to the field.

Reply:
you're right, I removed the method and reverted back to a default setter for user reaction.